### PR TITLE
pypi: Refactor pypi

### PR DESCRIPTION
**Pull request by georgebisbas**

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,10 +1,3 @@
 [build-system]
-requires = [
-    "setuptools>=42",
-    "wheel"
-]
-
-[tool.pyright]
```
Comment (webmiche) [inline pyproject.toml:7]: Do you really want/need to delete this?

**Review by webmiche — COMMENTED** [review]
I trust your call here. There is onl one thing I am uncertain about, but I am fine with doing it like this.

**Comment by math-fehr** [comment]
Nice, thanks a lot!
I tried to update the version yesterday, but I had no idea on how to update `versioneer` version, so I'm happy that we have this now!
We have an `update_pypi.sh` script to "manually" update the package. Do you think we should keep it, or should we remove it once the CI is working?

**Comment by georgebisbas** [comment]
I think its best to automate everything, will now merge: https://github.com/xdslproject/xdsl/pull/218
to check If I did right and will come back

**Pull request closed without merging**